api/Dto/Account/RegisterDto.cs

```diff
@@ -1,23 +1,24 @@
 using System.ComponentModel.DataAnnotations;
+using api.DataAnnotations;
 
 namespace api.Dto.Account
 {
     public class RegisterDto
     {
-        [Required]
+        [Required, StudentIdValidation]
         public int StudentId { get; set; }
         
         [Required]
         [EmailAddress]
         public required string Email { get; set; }
 
-        [Required]
+        [Required, RegularExpression(@"^[А-ЯЁ][а-яё]+(?:\s[А-ЯЁ][а-яё]+){2}$", ErrorMessage = "Invalid username")]
```

Comment:
```suggestion
        [Required, RegularExpression(@"^[А-ЯЁ][а-яё]+(?:\s[А-ЯЁ][а-яё]+){2}$", ErrorMessage = "Incorrect username")]
```

Старайся не использовать `Invalid` нигде в видимых сообщениях для пользователя. А то для иностранного сегмента это может звучать грубо :)